app/controllers/spree/orders_controller_decorator.rb

```diff
@@ -138,11 +138,19 @@ def update_totals_and_taxes
     @order.shipment.ensure_correct_adjustment_with_included_tax if @order.shipment
   end
 
-  # Sets the adjustments to open to perform the block's action and closes them back again
+  # Sets the adjustments to open to perform the block's action and restores
+  # their state to whatever the they had.
   def with_open_adjustments
+    previous_states = @order.adjustments.each_with_object({}) do |adjustment, hash|
+      hash[adjustment.id] = adjustment.state
+    end
     @order.adjustments.each(&:open)
+
     yield
-    @order.adjustments.each(&:close)
+
+    @order.adjustments.each_with_index do |adjustment, index|
```

Comment:
- You are not using the `index` here.
- What if something adds an adjustment? We should only restore adjustments that are in our states hash.